### DISCUSSION TITLE
[vcpkg upgrade] upgrade does not honor previous vcpkg install <port> --editable feature

### DISCUSSION THREAD

**Discussion by StarGate-One** [microsoft/vcpkg#17804] [Ideas]
**Describe the bug**
A port built using vcpkg install <port> --editable is not honored or rebuilt editable when using vcpkg upgrade
The vcpkg upgrade creates a src.clean directory then deletes all the ports source/build directories in `<vcpkg-root>\buildtrees\<port>` not just the .clean directory...

**Environment**
- OS: [Windows 10 / Linux both WSL2 and Hyper-V ]
- Compiler: msvc and gcc

**To Reproduce**
1. ./vcpkg install <port-1> <port-2> --editable
2. vcpkg root buildtrees port src/release/debug build/intermediate files and logs exist
2. port is then upgraded
3. ./vcpkg upgrade --no-dry-run 
4. port is upgrade but all source/build directories are gone, only a few logs are left not giving SOX auditors what they require.

**Expected behavior**
1. I believe `.\vcpkg upgrade --no-dry-run` should either honor the previous build process ie delete the source/build directories then recreate them if the --editable was used to previously create the port or add an --editable flag to the vcpkg upgrade process.
2. Also, please would it be possible to create an environment variable the vcpkg install/upgrade process can use storing the desired editable option, maybe even add it to the VCPKG_FEATURE_FLAGS???
3. Note: We need the buildtrees and all intermediate files created to not be deleted as our SOX compliancy officer requires all source, intermediate and end files for audit purposes.
4. I understand the need to want to clean up afterwards, but SOX regulations dealing with financial processing is heavily audited and controlled - ie, payroll, accounts payable, account receivable, general ledger, etc.
5. I have already had to modify almost all ports using vcpkg_install_msbuild() function adding the SKIP_CLEAN to not delete build/intermediate files.
6. There needs to be a global option we can use so no matter which way a port is built, its build/intermediate files are not deleted, whether it is the initial install or upgrade process.
Thank you.

**Failure logs** 
None

**Additional context**

**Comment by strega-nil** [comment]
It'd be nice to add the `--editable` flag to `upgrade`, you're right.

**Comment by StarGate-One** [comment]
@strega-nil 
@JackBoosY 
Thank you, but is there an estimate on when this could take effect?
Reason I ask, as until the editable option is available, the `vcpkg upgrade` option is now worthless and I have to manually create scripts to remove and reinstall the ports upgraded and any with dependencies on those ports with the selected features I require. 
A lot of manual work which did not exist before the editable option was added.

**Comment by StarGate-One** [comment]
Closing, no longer an issue for me as we have chosen to no long use vcpkg.

**Comment by JackBoosY** [comment]
Reopen this issue since it's not resolved.

**Comment by StarGate-One** [comment]
Even though I started using vcpkg again, the issue can be solved by having a script (easily done in cmd, pwsh, bash depending on OS) to remove the ports requiring upgrade via `vcpkg remove <ports> <ports>...` then follow-up to re-install the ports just removed, adjusting for any new ports/features via `vcpkg install <ports> <ports> ...`  :)
